### PR TITLE
Beats dashboard should use custom index patterns when `setup.dashboards.index` is set

### LINKED ISSUE #21232: Custom index patterns not fully handled by dashboards
There are two parts that are not being updated when setting a custom index pattern via `setup.dashboards.index`.

1. Timelion expressions containing index patterns are not updated. [example](https://github.com/elastic/beats/blob/ad823eca4cc74439d1a44351c596c12ab51054f5/x-pack/filebeat/module/netflow/_meta/kibana/7/dashboard/filebeat-netflow-autonomous-systems.json#L221-L227)
1. `index-pattern` references from dashboard objects are not updated. [example](https://github.com/elastic/beats/blob/913b2548b3fef7498bc7e6e7be5c6969a592f5fc/x-pack/filebeat/module/netflow/_meta/kibana/7/dashboard/filebeat-netflow-geo-location.json#L100-L105)

Regarding timelion, there's no code to update the `expression` with the new index pattern. I think that would belong in

https://github.com/elastic/beats/blob/ad823eca4cc74439d1a44351c596c12ab51054f5/libbeat/dashboards/modify_json.go#L126-L155

And for the dashboard objects we need to update the `references` where type=index-pattern. I think that would belong in

https://github.com/elastic/beats/blob/ad823eca4cc74439d1a44351c596c12ab51054f5/libbeat/dashboards/modify_json.go#L167

Relates
- https://github.com/elastic/beats/issues/20703

### PR DESCRIPTION
**Pull request by kvch**
## What does this PR do?

This PR adds a new check to `make check` to make sure all dashboards can handle custom index names. It is the responsibility of the dashboard author to keep the dashboard useable if `setup.dashboards.index` is configured.

The PR contains additional fixes to make sure existing dashboards can be used with custom index names. Also, the test coverage was increased as it is a quite big PR.

The PR also decodes a few fields to make dashboards easier to review.

## Why is it important?

The bug prevents users from setting custom index names and use our dashboards at the same time.

## Checklist

- [x] My code follows the style guidelines of this project
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

Closes #21232

### CODE REVIEW

File: x-pack/filebeat/module/cyberarkpas/_meta/kibana/7/dashboard/Filebeat-cyberarkpas-audit.json [hunk 1]
```diff
@@ -972,7 +972,7 @@
...
...S_PEW_PEW\",\"applyGlobalQuery\":true,\"applyGlobalTime\":true},\"style\":{\"type\":\"VECTOR\",\"properties\":{\"icon\":{\"type\":\"STATIC\",\"options\":{\"value\":\"marker\"}},\"fillColor\":{\"type\":\"STATIC\",\"options\":{\"color\":\"#54B399\"}},\"lineColor\":{\"type\":\"STATIC\",\"options\":{\"color\":\"#6092C0\"}},\"lineWidth\":{\"type\":\"DYNAMIC\",\"options\":{\"minSize\":1,\"maxSize\":8,\"field\":{\"name\":\"doc_count\",\"origin\":\"source\"},\"fieldMetaOptions\":{\"isEnabled\":true,\"sigma\":3}}},\"iconSize\":{\"type\":\"STATIC\",\"options\":{\"size\":6}},\"iconOrientation\":{\"type\":\"STATIC\",\"options\":{\"orientation\":0}},\"labelText\":{\"type\":\"STATIC\",\"options\":{\"value\":\"\"}},\"labelColor\":{\"type\":\"STATIC\",\"options\":{\"color\":\"#000000\"}},\"labelSize\":{\"type\":\"STATIC\",\"options\":{\"size\":14}},\"labelBorderColor\":{\"type\":\"STATIC\",\"options\":{\"color\":\"#FFFFFF\"}},\"symbolizeAs\":{\"options\":{\"value\":\"circle\"}},\"labelBorderSize\":{\"options\":{\"size\":\"SMALL\"}}},\"isTimeAware\":true},\"id\":\"9c450fbf-b009-4b53-9810-2f47ca8dcfa8\",\"minZoom\":0,\"maxZoom\":24,\"alpha\":0.75,\"visible\":true,\"type\":\"VECTOR\",\"joins\":[]}]",
```
Comment (kvch) [inline x-pack/filebeat/module/cyberarkpas/_meta/kibana/7/dashboard/Filebeat-cyberarkpas-audit.json:975]: filebeat-* was renamed to Filebeat index to make the label more generic.

File: x-pack/filebeat/module/aws/_meta/kibana/7/map/0edf0640-3e7e-11ea-bb0a-69c3ca1d410f.json [hunk 2]
```diff
@@ -1,8 +1,154 @@
...
-        "mapStateJSON": "{\"zoom\":3.9,\"center\":{\"lon\":13.666,\"lat\":50.97903},\"timeFilters\":{\"from\":\"now-15m\",\"to\":\"now\"},\"refreshConfig\":{\"isPaused\":false,\"interval\":0},\"query\":{\"query\":\"\",\"language\":\"kuery\"},\"filters\":[{\"meta\":{\"index\":\"filebeat-*\",\"alias\":null,\"negate\":false,\"disabled\":false,\"type\":\"phrase\",\"key\":\"fileset.name\",\"value\":\"elb\",\"params\":{\"query\":\"elb\"}},\"query\":{\"match\":{\"fileset.name\":{\"query\":\"elb\",\"type\":\"phrase\"}}},\"$state\":{\"store\":\"appState\"}}],\"settings\":{\"autoFitToDataBounds\":false}}",
```
Comment (jsoriano) [inline x-pack/filebeat/module/aws/_meta/kibana/7/map/0edf0640-3e7e-11ea-bb0a-69c3ca1d410f.json:5]: :+1:

File: libbeat/dashboards/modify_json.go [hunk 3]
```diff
@@ -208,6 +241,140 @@ func ReplaceIndexInDashboardObject(index string, content []byte) []byte {
...
+			panel["embeddableConfig"] = config
+			panelsJSON[i] = panel
+		}
+	}
+	return panelsJSON
+}
+
+func replaceIndexInParamControls(logger *logp.Logger, index string, params map[string]interface{}) map[string]interface{} {
+	if contolsList, ok := params["controls"].([]interface{}); ok {
+		for i, ctrl := range contolsList {
```
Comment (jsoriano) [inline libbeat/dashboards/modify_json.go:325]:
```suggestion
	if controlsList, ok := params["controls"].([]interface{}); ok {
		for i, ctrl := range controlsList {
```

File: x-pack/filebeat/module/cyberarkpas/_meta/kibana/7/dashboard/Filebeat-cyberarkpas-audit.json (hunk 1 above)
Comment (jsoriano) [reply x-pack/filebeat/module/cyberarkpas/_meta/kibana/7/dashboard/Filebeat-cyberarkpas-audit.json:975]: Shouldn't this be decoded after https://github.com/elastic/beats/pull/27901/files#diff-69c90b10fb317dfe595ea552d5701edae6a612bd43c7514a06d435e2a3a601f3R34 ?

File: x-pack/filebeat/module/cyberarkpas/_meta/kibana/7/dashboard/Filebeat-cyberarkpas-audit.json [hunk 4]
```diff
@@ -972,7 +972,7 @@
...
                         "mapStateJSON": "{\"zoom\":1.24,\"center\":{\"lon\":-49.38072,\"lat\":7.87497},\"timeFilters\":{\"from\":\"now-15w\",\"to\":\"now\"},\"refreshConfig\":{\"isPaused\":true,\"interval\":0},\"query\":{\"query\":\"\",\"language\":\"kuery\"},\"filters\":[],\"settings\":{\"autoFitToDataBounds\":false,\"backgroundColor\":\"#ffffff\",\"disableInteractive\":false,\"disableTooltipControl\":false,\"hideToolbarOverlay\":false,\"hideLayerControl\":false,\"hideViewControl\":false,\"initialLocation\":\"LAST_SAVED_LOCATION\",\"fixedLocation\":{\"lat\":0,\"lon\":0,\"zoom\":2},\"browserLocation\":{\"zoom\":2},\"maxZoom\":24,\"minZoom\":0,\"showScaleControl\":false,\"showSpatialFilters\":true,\"spatialFiltersAlpa\":0.3,\"spatialFiltersFillColor\":\"#DA8B45\",\"spatialFiltersLineColor\":\"#DA8B45\"}}",
```
Comment (jsoriano) [inline x-pack/filebeat/module/cyberarkpas/_meta/kibana/7/dashboard/Filebeat-cyberarkpas-audit.json:976]: And this.

**Review by jsoriano — COMMENTED** [review]
:+1:

File: x-pack/filebeat/module/cyberarkpas/_meta/kibana/7/dashboard/Filebeat-cyberarkpas-audit.json (hunk 4 above)
Comment (kvch) [reply x-pack/filebeat/module/cyberarkpas/_meta/kibana/7/dashboard/Filebeat-cyberarkpas-audit.json]: Decoded. Also, decoded other files...

**Review by jsoriano — APPROVED** [review]

**Pull request merged**